pages/company/visualizations/unqiue_domains.py

```diff
@@ -0,0 +1,249 @@
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    # get date value for first contribution
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+    df = df.sort_values(by="created", axis=0, ascending=True)
+    base = df.iloc[0]["created"]
+
+    date_picker = (
+        dcc.DatePickerRange(
+            id=f"{PAGE}-date-picker-range-{VIZ_ID}",
+            min_date_allowed=base,
+            max_date_allowed=dt.date.today(),
+            clearable=True,
+        ),
+    )
+    return date_picker
+
+
+# callback for Company Affiliation by Github Account Info graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"{PAGE}-{paramter_2}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-{paramter_1}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "start_date"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "end_date"),
+    ],
+    background=True,
+    prevent_initial_call=True,
+)
+def unique_domains_graph(repolist, checks, num, start_date, end_date):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, checks, num, start_date, end_date)
+
+    fig = create_figure(df)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, checks, num, start_date, end_date):
+
+    # convert to datetime objects rather than strings
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+
+    # order values chronologically by COLUMN_TO_SORT_BY date
+    df = df.sort_values(by="created", axis=0, ascending=True)
+
+    # filter values based on date picker
+    if start_date is not None:
+        df = df[df.created >= start_date]
+    if end_date is not None:
+        df = df[df.created <= end_date]
+
+    # creates list of unique emails and flattens list result
+    emails = df.email_list.str.split(" , ").explode("email_list").unique().tolist()
+
+    # remove any entries not in email format
+    emails = [x for x in emails if "@" in x]
+
+    # creates list of email domains from the emails list
+    email_domains = [x[x.rindex("@") + 1 :] for x in emails]
+
+    # creates df of domains and counts
+    df = pd.DataFrame(email_domains, columns=["domains"]).value_counts().to_frame().reset_index()
```

Comment:
If you've taken all unique values then shouldn't the value_counts() be = 1?

Reply:
Took all of the unique emails, not domains so now that it is in "gmail.com" format its counting the domains not the emails